docs/source/developers/guide/step_by_step/pr_and_github.rst

```diff
@@ -127,6 +127,49 @@ If all is set, you can make the pull request!
       and other additional information :ref:`pull_request_and_review`
       section.
 
+Continuous Integration (CI)
+---------------------------
+
+Continuous integration (CI) is an automated way to assert the code
+change made by a specific pull request before it gets merged or
+integrated into the main repository of the project.
+
+Once the pull request is created, the CI will trigger checks on the
+code. Depending on what part of the code was changed (documentation,
+C++ or other languages for example) the CI will choose the corresponding
+checks to be made.
+
+You will see checks running at the bottom of the pull request page
+on GitHub. In case of an error, click on the details and research the cause
+of the failing build.
+
+.. figure:: ci_process_docs.jpeg
+   :scale: 60 %
+   :alt: CI window showing the status of the code checks
+         in case of changes made to the documentation.
+
+   CI checks for changes made to the documentation.
+
+.. figure:: ci_process_python.jpeg
+   :scale: 58 %
+   :alt: CI window showing the status of the code checks
+         in case of changes made to the python files
+
+   CI checks for changes made to the python files.
+
+.. figure:: ci_process_r.jpeg
+   :scale: 58 %
+   :alt: CI window showing the status of the code checks
+         in case of changes made to the R files.
+
+   CI checks for changes made to the R files.
+
+Besides the CI jobs that check the changes in GitHub repository
+(opening or merging of a pull request) we also use CI for nightly
+builds and releases of the Apache Arrow library.
+
```

Comment:
How about adding a bit here about triggering jobs on-demand via `@github-actions crossbow submit ...`?

Reply:
Tried to add info that would be, maybe, needed for a first time contributor. What do you think?